src/components/editor/setupLanguages.ts

```diff
@@ -0,0 +1,65 @@
+import { Registry } from 'monaco-textmate';
+import { wireTmGrammars } from 'monaco-editor-textmate';
+import * as monaco from 'monaco-editor';
+import cssDefinition from './languages/css.tmLanguage.json?url';
+import tsxDefinition from './languages/TypeScriptReact.tmLanguage.json?url';
+
+const grammars = new Map();
+grammars.set('css', 'source.css');
+// grammars.set('javascript', 'source.js');
+grammars.set('javascript', 'source.js.jsx');
+// grammars.set('jsx', 'source.js.jsx');
+// grammars.set('tsx', 'source.tsx');
+// grammars.set('typescript', 'source.ts');
+grammars.set('typescript', 'source.tsx');
+
+const inverseGrammars: Record<string, string> = {
+  'source.css': 'css',
+  'source.js': 'jsx',
+  // 'source.js': 'javascript',
+  'source.js.jsx': 'jsx',
```

Comment:
Is this used?

Reply:
TBF we can just remove it. jsx and tsx has different grammars.